### PR TITLE
feat: [IDP-1949][IDP-1963]: Audit trails for plugin enable disbale and configuration screen updated

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57503)

### CODE REVIEW

**Comment by devesh-harness** [comment]
trigger messagecheck

**Review by vikyathharekal — DISMISSED** [review]
> Reviewed 17 of 20 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57503)* status: 20 file(s) to review. 1 of 2 approvals obtained. 

___
*[`idp-service/src/main/java/io/harness/idp/envvariable/service/BackstageEnvVariableServiceImpl.java` line 576 at r2](https://reviewable.io/reviews/harness/harness-core/57503#-NoAL6tIDKVzW6JmwA4l:-NoAL6tIDKVzW6JmwA4m:b5j15nd) ([raw file](https://github.com/harness/harness-core/blob/bb40994b27fb27a80d1a74e38a49ba5d7b72dd23/idp-service/src/main/java/io/harness/idp/envvariable/service/BackstageEnvVariableServiceImpl.java#L576)):*
> ```Java
>     return Failsafe.with(transactionRetryPolicy).get(() -> transactionTemplate.execute(status -> {
>       envVariables.forEach(envVariable -> {
>         if (envVarsFromDB.keySet().contains(envVariable.getEnvName())) {
> ```

nit: We can do `envVarsFromDB.keySet().contains` instead



_Code quote:_
```Java
envVarsFromDB.keySet().contains
```

**Review by mankrit-singh — APPROVED** [review]

**Comment by devesh-harness** [comment]
trigger uts

**Review by sandeepa-kv-jois — APPROVED** [review]

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut6

**Comment by devesh-harness** [comment]
trigger ut9

**Review by vikyathharekal — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57503)* status: 25 file(s) to review. 1 of 2 approvals obtained. 

___
*[`idp-service/contracts/openapi/v1/openapi.yaml` line 601 at r4](https://reviewable.io/reviews/harness/harness-core/57503#-NoBt5S7EJ13s24c_W1e:-NoBt5S7EJ13s24c_W1f:b-c5zvjx) ([raw file](https://github.com/harness/harness-core/blob/0d433bba1c9690fe29c33ce5182e992367a35508/idp-service/contracts/openapi/v1/openapi.yaml#L601)):*
> ```YAML
>       parameters:
>         - $ref: '#/components/parameters/AccountHeader'
>         - $ref: '#/components/parameters/PluginName'
> ```

Please inform Jenil that this needs to be sent for custom plugin delete in UI

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut2, ut5, ut6, ut7

File: idp-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -4028,7 +4030,7 @@ components:
       schema:
         type: string
     PluginName:
-      name: PluginId
+      name: plugin-name
```
Comment (mankrit-singh) [inline idp-service/contracts/openapi/v1/openapi.yaml:4033]: casing

Comment (devesh-harness) [reply idp-service/contracts/openapi/v1/openapi.yaml]: done

**Review by mankrit-singh — APPROVED** [review]

**Comment by devesh-harness** [comment]
trigger uts

**Review by vikyathharekal — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57503#-:-NoCsFht9fizYIn6vzKA:bnfp4nl)* status: 25 file(s) to review. 1. unresolved discussions.

**Comment by devesh-harness** [comment]
trigger review

**Pull request merged**